### ISSUE TITLE
data_source.php - sql error

### ISSUE THREAD

**Issue by xmacan** [Cacti/cacti#1639]
Management - > Data source -> choose any host and click any data source name, the log displays:

2018/05/16 09:53:16 - DBCALL ERROR: SQL Assoc Failed!, Error:1064, SQL:'SELECT  dspc.consolidation_function_id AS cf,  dsp.x_files_factor AS xff,  dspr.steps AS steps,  dspr.rows AS rows  FROM data_source_profiles AS dsp  INNER JOIN data_source_profiles_cf AS dspc  ON dsp.id=dspc.data_source_profile_id  INNER JOIN data_source_profiles_rra AS dspr  ON dsp.id=dspr.data_source_profile_id  WHERE dsp.id = ?  ORDER BY dspc.consolidation_function_id, dspr.steps'
2018/05/16 09:53:16 - DBCALL ERROR: SQL Assoc Failed!, Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'rows  FROM data_source_profiles AS dsp  INNER JOIN data_source_profiles_cf AS ds' at line 1
2018/05/16 09:53:16 - CMDPHP SQL Backtrace: (/data_sources.php: 77 ds_edit)(/data_sources.php: 1036 rrdtool_cacti_compare)(/lib/rrd.php: 2459 db_fetch_assoc_prepared)(/lib/database.php: 401 cacti_debug_backtrace

Version 1.1.36, mariaDB 10.2.13

**Comment by xmacan** [comment]
I found problem! ROWS is no prohibited word in mariaDB:
https://mariadb.com/kb/en/library/reserved-words/

ROWS | Added in MariaDB 10.2.4

**Comment by xmacan** [comment]
I made pull request with fix.

**Comment by netniV** [comment]
Good find and thanks for the patch.  However, I have made a suggestion for an alternative fix as I think it reduces the needed changes.  It does mean that we need to look through all code to see if that word is used and correct were applicable.

**Comment by xmacan** [comment]
I updated pull request as you sugested. I fixed the same problem in user_admin.php

**Comment by TakeMeNL** [comment]
have this issue with 1.1.38 and mariadb 10.3.9. Manually patched for now...

**Comment by janpieters74** [comment]
Dear,
What manual patch did you apply ?
the moment I updated to version 1.1.38_7 and used MariaDB 10.2.20 the problems started (no graphing anymore, historical data is still graphed).

This is the erro I get: BCALL ERROR: SQL Assoc Failed!, Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'rows FROM data_source_profiles AS dsp INNER JOIN data_source_profiles_cf AS ds' at line 1

**Comment by TakeMeNL** [comment]
> Dear,
> What manual patch did you apply ?
> the moment I updated to version 1.1.38_7 and used MariaDB 10.2.20 the problems started (no graphing anymore, historical data is still graphed).
> 
> This is the erro I get: BCALL ERROR: SQL Assoc Failed!, Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'rows FROM data_source_profiles AS dsp INNER JOIN data_source_profiles_cf AS ds' at line 1

https://github.com/Cacti/cacti/commit/80f7ccd86b45318f451ac2f981cc771d24eb1d1d

**Comment by netniV** [comment]
Yes, this is due to reserved words being expanded by MySQL and having a column of that same name is now required to be in quotes.  1.2 should work with issue when it's released.

**Comment by janpieters74** [comment]
Any idea about the releasedate?

Op do 27 dec. 2018 13:57 schreef Mark Brugnoli-Vinten <
notifications@github.com:
